rom/dev/tests/smoke_test.rs

```diff
@@ -0,0 +1,24 @@
+// Licensed under the Apache-2.0 license
+
+use caliptra_builder::{ImageOptions, APP_WITH_UART, FMC_WITH_UART, ROM_WITH_UART};
+use caliptra_hw_model::{HwModel, InitParams};
+
+#[test]
+fn smoke_test() {
+    let rom = caliptra_builder::build_firmware_rom(&ROM_WITH_UART).unwrap();
+    let image = caliptra_builder::build_and_sign_image(
+        &FMC_WITH_UART,
+        &APP_WITH_UART,
+        ImageOptions::default(),
+    )
+    .unwrap();
+    let mut hw = caliptra_hw_model::create(InitParams {
+        rom: &rom,
+        ..Default::default()
+    })
+    .unwrap();
```

Comment:
Should we step until flow_status.ready_for_fw  is set?

Reply:
I want upload_firmware() to do that.

Reply:
Removed from this PR.